scripts/ingests/ingest_utils.py

```diff
@@ -0,0 +1,782 @@
+        else:
+            msg = f"{i}: unexpected condition encountered ingesting {source}"
+            logger.error(msg)
+            raise SimpleError(msg)
+
+        # Try to add the source to the database
+        try:
+            db.Sources.insert().execute(source_data)
+            n_added += 1
+            msg = f"Added {str(source_data)}"
+            logger.debug(msg)
+        except sqlalchemy.exc.IntegrityError:
+            if ma.is_masked(source_data[0]['reference']):  # check if reference is blank
+                msg = f"{i}: Skipping: {source}. Discovery reference is blank. \n"
+                msg2 = f"\n {str(source_data)}\n"
+                logger.warning(msg)
+                logger.debug(msg2)
+                n_skipped += 1
+                if raise_error:
+                    raise SimpleError(msg + msg2)
+                else:
+                    continue
+            elif db.query(db.Publications).filter(db.Publications.c.name == references[i]).count() == 0:
+                # check if reference is in Publications table
+                msg = f"{i}: Skipping: {source}. Discovery reference {references[i]} is not in Publications table. \n" \
+                      f"(Add it with add_publication function.) \n "
+                msg2 = f"\n {str(source_data)}\n"
+                logger.warning(msg)
+                logger.debug(msg2)
+                n_skipped += 1
+                if raise_error:
+                    raise SimpleError(msg + msg2)
+                else:
+                    continue
+            # try reference without last letter e.g.Smit04 instead of Smit04a
+            # elif source_data[0]['reference'][-1] in ('a', 'b'):
+            #    source_data[0]['reference'] = references[i][:-1]
+            #    try:
+            #        db.Sources.insert().execute(source_data)
+            #        n_added += 1
+            #        msg = f"Added \n {str(source_data)}"
+            #        logger.debug(msg)
+            #    except sqlalchemy.exc.IntegrityError:
+            #        msg = f"Skipping {source} "
+            #        msg2 = f"\n {str(source_data)} " \
+            #               f"\n Discovery reference may not exist in the Publications table. " \
+            #               "(Add it with add_publication function.) \n "
+            #        logger.warning(msg)
+            #        logger.debug(msg2)
+            #        n_skipped += 1
+            #        if raise_error:
+            #            raise SimpleError(msg + msg2)
+            #        else:
+            #            continue
+            else:
+                msg = f"{i}: Skipping: {source}. Not sure why."
+                msg2 = f"\n {str(source_data)} "
+                logger.warning(msg)
+                logger.debug(msg2)
+                n_skipped += 1
+                if raise_error:
+                    raise SimpleError(msg+msg2)
+                else:
+                    continue
```

Comment:
```suggestion
                continue
```
technically don't need the else: continue, as the continue will always activate if the raise error is False. Doesn't matter though!

Reply:
I'm gonna leave it. :)